File: Beecrowd/2 - Ad-Hoc/1026 - ~/1031 - Power Crisis/C++/1031_PowerCrisis.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int regions;
    while (cin >> regions && regions) {
        int randomNumber = 1;

        while (true) {
            vector<int> v(regions);

            iota(v.begin(), v.end(), 1);

            int i = 0;
            while (v.size() > 1) {
                v.erase(v.begin() + i);
                i = (i + randomNumber - 1) % v.size();
            }

            if (v[0] == 13) {
                cout << randomNumber << endl;
                break;
            }

            randomNumber++;
        }
    }
    
    return 0;
}

File: Beecrowd/2 - Ad-Hoc/1026 - ~/1031 - Power Crisis/1031_PowerCrisis.py
while True:
    regions = int(input())
    if regions == 0:
        break

    randomNumber = 1

    while True:
        v = list(range(1, regions + 1))
        
        i = 0
        while len(v) > 1:
            del v[i]
            i = (i + randomNumber - 1) % len(v)

        if v[0] == 13:
            print(randomNumber)
            break

        randomNumber += 1
